File: api/create_file_move_paths.ts
import type { VercelRequest, VercelResponse } from '@vercel/node';
import * as dotenv from 'dotenv';
import { MongoClient } from 'mongodb';

interface FileMovePaths {
	current: string;
	new: string;
	name?: string;
}

export default (request: VercelRequest, response: VercelResponse) => {
	dotenv.config();
	const client = new MongoClient(process.env.MONGODB_URI || "");
	client.connect().then(() => {
		const collection = client.db("data").collection("file_move_paths");
		try {
			let body = request.body;
			let tmp: FileMovePaths[] = [];
			for (let i = 0; i < body.data.length; i++) {
				let data: FileMovePaths = {
					"current": body.data[i]["current"],  // required current path
					"new": body.data[i]["new"]  // required new path
				}
				if ("name" in body.data[i]) {
					data.name = body.data[i]["name"];  // optional name
				}
				tmp.push(data);
			}
			collection.insertMany(tmp);
			response.status(200).send(request.body);
		} catch (err) {
			response.status(400).json({ error: "bad body format" });
		}
	}).catch((err) => {
		response.status(500).json({ error: "could not connect to database" });
	}).finally(() => {
		client.close();
	});
};

File: api/create_file_move_paths.py
from http.server import BaseHTTPRequestHandler
import os
from pymongo import MongoClient
import json
from flask import Flask, Response
app = Flask(__name__)

@app.route('/', defaults={'path': ''}, methods=['PUT'])
@app.route('/<path:path>', methods=['PUT'])
def catch_all(path):
    from flask import request
    if request.headers.get('Content-Type') != 'application/json': return Response("Content-Type must be application/json", mimetype='text/plain', status=400)
    body = request.json
    data = [{
        "name": i["name"] if "name" in i.keys() else None,
        "new": i["new"],
        "curret": i["current"]
    } for i in body["data"]]
    MongoClient(os.environ['MONGODB_URI']).data.file_move_paths.insert_many([{j:i[j] for j in i.keys() if i[j] != None} for i in data])
    return Response("OK")
